src/identity/with-authentication.ts

```diff
@@ -54,7 +77,17 @@ export function withAuthentication({ roles = [], permissions = [] }: IWithAuthen
 
       // Store the decoded user on the request
       req.user = user;
-      req.user.id = user.sub; // The subject of the token (OpenID token) is saved on the req.user as well for easier readability
+      req.user.id = ''
+
+      switch (req.user.type) {
+        case tokenTypes.UserToken:
+          req.user.id = user.sub; // The subject of the token (OpenID token) is saved on the req.user as well for easier readability
+          break;
+        case tokenTypes.UserApiToken:
```

Comment:
We should add for the tenant api token as well.
Not a blocker but let's remember it